### PR TITLE
URLQuery with index slice keys

### PR DESCRIPTION
**Pull request by earncef**
Certain APIs (I believe those written in PHP) seem to be unable to parse correctly URLQuery() result containing slices generated by URLValuesSliceKeySuffix="" or URLValuesSliceKeySuffix="[]".

The solution is to pass the slices with an index so that they are read correctly. 

For instance, in order to parse slices of the form correctly: 
objx.Map{"mapSlice": []objx.Map{objx.Map{"age": 40, "sex": "male"}, objx.Map{"height": 152}}}
the query needs to be: 
mapSlice[0][age]=40&mapSlice[0][sex]=male&mapSlice[1][height]=152

Currently these two forms are possible which don't work:
mapSlice[age]=40&mapSlice[sex]=male&mapSlice[height]=152
mapSlice[][age]=40&mapSlice[][sex]=male&mapSlice[][height]=152

I've modified the code to check for URLValuesSliceKeySuffix="[i]" and use slice indexes when present

### CODE REVIEW

**Comment by earncef** [comment]
Added tests wrt #71 but I will write additional tests separately.

**Comment by Hanzei** [comment]
@earncef Could give this a rebase?

**Comment by earncef** [comment]
Done

File: conversions.go [hunk 1]
```diff
@@ -131,25 +148,42 @@ func (m Map) parseURLValues(queryMap Map, vals url.Values, key string) {
 				m.parseURLValues(New(v), vals, key+"["+k+"]")
 			}
 		case val.IsMSISlice():
```
Comment (Hanzei) [inline conversions.go]: This case isn't covered by any testcases. Mind adding some? I think you can reuse the `val.IsObjxMapSlice()` ones

**Comment by earncef** [comment]
I agree. What would you consider the case for error though? Not being one of "" or "[]" or "[I]"?

File: conversions.go (hunk 1 above)
Comment (earncef) [reply conversions.go]: Due to a previous pull request, this part is no longer necessary. Updated the code and added tests.

**Comment by Hanzei** [comment]
> I agree. What would you consider the case for error though? Not being one of "" or "[]" or "[I]"?

Correct. I would limit `SetURLValuesSliceKeySuffix` to these three values and throw an error for every other.

**Comment by earncef** [comment]
Done

**Pull request merged**